src/function/table.rs

```diff
@@ -46,12 +47,31 @@ struct TableOffset {
     off: Offset,
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct Row {
+    pub(crate) input: Input,
+    pub(crate) output: TupleOutput,
+    pub(crate) unextractable: bool,
+    pub(crate) subsumed: bool,
+}
+
+impl Row {
+    fn new(input: Input, output: TupleOutput, unextractable: bool, subsumed: bool) -> Row {
+        Row {
+            input,
+            output,
+            unextractable,
+            subsumed,
+        }
+    }
+}
+
 #[derive(Default, Clone)]
 pub(crate) struct Table {
     max_ts: u32,
     n_stale: usize,
     table: RawTable<TableOffset>,
-    pub(crate) vals: Vec<(Input, TupleOutput)>,
+    pub(crate) vals: Vec<Row>,
```

Comment:
I wasn't sure if the unextractable and subsumed metadata should be on the input or output, so I just put them on the row itself and made it a struct.